src/redux/session.js

```diff
@@ -31,27 +41,29 @@ module.exports.setSessionError = function (error) {
     };
 };
 
-module.exports.setSession = function (session) {
+module.exports.setSession = function (status, results) {
     return {
         type: Types.SET_SESSION,
-        session: session
+        session: {'status': status,'results': results}
     };
 };
 
 module.exports.refreshSession = function () {
     return function (dispatch) {
+        dispatch(module.exports.setSession(module.exports.Status.NOT_FETCHED, {}));
```

Comment:
A few things here:
* The status at the beginning of the api call should be fetching to reflect that the api call has started but not completed. `NOT_FETCHED` should be the initial state before anything has touched the session
* I think we should split the two parts of the state and handle them separately so we can retain the session data until we have the refreshed content. E.g., at the beginning just update the status and don't change the session data, then when you've fetched it, dispatch two actions, one to update the status and one to update the session data.
* Splitting the actions to update the status and session data separately will require changing the reducer so it only updates either the session data or the status. When you do that you should not do what is most intuitive and change the state object passed into the reducer — the state object should be treated as immutable, so you should use `lodash.defaultsDeep` to return a new state object with the appropriate values. e.g. `case Types.SET_SESSION_STATUS: return defaultsDeep({}, {status: action.status}, state)`.  Sorry this is kind of overcomplicated for the task at hand, but it's part of the way redux works that isn't obvious if you're just looking at the existing code here.